Extensions/xkit_preferences.js

```diff
@@ -96,7 +97,13 @@ XKit.extensions.xkit_preferences = new Object({
 				}
 
 				if (nav && !nav.closest(drawerContent)) {
-					nav.append(button);
+					const primaryNavItems = [...nav.children].filter(
+						el => el.querySelector(startChildWrapper) || (el.matches('ul') && el.previousElementSibling.querySelector(startChildWrapper))
```

Comment:
is there a more obvious / easy-to-understand way we could write this, I'm having trouble getting my head around the logic of it

Reply:
WDYT about just saying "Put it after the settings button if it exists"

Reply:
yeah, thinking about this more it seems like the actual logic is "Put this after the last collapsible menu item" which just seems more fragile and less clear from a user perspective compared to just putting it after a specific nav item

Reply:
Re: putting it after settings, there are two problems:
 - The settings and account buttons are followed by collapsible menus, which are separate DOM elements from their corresponding buttons... only in the >1162px range. So one can say neither "find this button and put it after" or "find this collapsible menu and put it after"—handling both correctly is required.
 - As noted in the quoted paragraph, "after settings" causes the button to be teleported above account. If we're willing to mix the button in along with the "proper" menu items, as it would be if it were right after settings, we should probably just put it above account. At that point it should probably have proper styling in the 1018-1162px range, but eh, someone can do that later, right*

*no one is probably going to do that later, I would imagine

I think the viable paths are "put it before account (I can go implement styling)" and "put it after the last collapsible menu item; this code can probably be made clearer in order to do that." I may not have thought of another one that exists, though.

Reply:
Oh, and the third path: "just ignore this and leave it at the end of the container/potentially manually whitelist some selectors which should go after it," which is perfectly reasonable.